addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/service-catalog/ec2-linux-instance.cfn.yml

```diff
@@ -41,6 +41,18 @@ Conditions:
   IamPolicyEmpty: !Equals [!Ref IamPolicyDocument, '{}']
 
 Resources:
+  InstanceRolePermissionBoundary:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: Permission boundary for EC2 instance role
+      ManagedPolicyName: !Join ['-', [Ref: Namespace, 'ec2-linux-permission-boundary']]
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Action:
+              - 's3:*'
```

Comment:
For all the workspaces roles- don't we need KMS access as well? Additionally, you might need to add AssumeRole access too since these roles will be assuming cross account roles. Do ensure that we test BYOB as well on these workspaces.

Reply:
Emmm, KMS does not seem to be part of the IAMRole permission. Good catch on the AssumeRole part! I missed the IamPolicyDocument part.

Reply:
KMS might be getting added as part of study permission updates. Without KMS how will these roles access MyStudy or Org studies?